src/librustc_metadata/encoder.rs

```diff
@@ -1435,6 +1458,8 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             def_path_table: def_path_table,
             impls: impls,
             exported_symbols: exported_symbols,
+            compiler_working_dir: tcx.sess.working_dir.to_string_lossy().into_owned(),
+            debug_prefix_map: debug_prefix_map,
```

Comment:
Sorry, I'm not familiar with the surrounding code. Does this mean the mapping is going to be recorded in the output? In that case, the output would contain some data that is dependent on the build-path. But ideally we'd like the output to be reproducible regardless of the build-path, which is one of the motivations for remapping debuginfo in the first place.

We ran into similar issues with GCC's `-fdebug-prefix-map` - if we set this on a distribution-wide level (`dpkg-buildflags` in Debian unstable does this at the moment) we find that some build tools will save `CFLAGS` etc into other parts of the build output (such as pkg-config files etc) which makes the overall output reproducible, even though GCC's own output was reproducible. Previously, GCC was also saving the value of this flag into the `DW_AT_producer` field, then they accepted a patch from us to specifically filter this flag out.